src/Generator/HtmlForPdfGenerator.php

```diff
@@ -2,41 +2,141 @@
 
 namespace SymfonyDocsBuilder\Generator;
 
+use Symfony\Component\DomCrawler\Crawler;
 use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Finder\Finder;
 use SymfonyDocsBuilder\BuildContext;
 
-/**
- * Class HtmlForPdfGenerator
- */
 class HtmlForPdfGenerator
 {
     use GeneratorTrait;
 
-    public function generateHtmlForPdf(
-        array $documents,
-        BuildContext $buildContext
-    ) {
+    public function generateHtmlForPdf(array $documents, BuildContext $buildContext) {
         $environments = $this->extractEnvironments($documents);
 
         $finder = new Finder();
         $finder->in($buildContext->getHtmlOutputDir())
             ->depth(0)
-            ->notName($buildContext->getParseOnly());
+            ->notName([$buildContext->getParseSubPath(), '_images']);
 
         $fs = new Filesystem();
         foreach ($finder as $file) {
             $fs->remove($file->getRealPath());
         }
 
-        $basePath  = sprintf('%s/%s', $buildContext->getHtmlOutputDir(), $buildContext->getParseOnly());
+        $basePath  = sprintf('%s/%s', $buildContext->getHtmlOutputDir(), $buildContext->getParseSubPath());
         $indexFile = sprintf('%s/%s', $basePath, 'index.html');
         if (!$fs->exists($indexFile)) {
-            throw new \InvalidArgumentException('File "%s" does not exist', $indexFile);
+            throw new \InvalidArgumentException(sprintf('File "%s" does not exist', $indexFile));
         }
 
+        // extracting all files from index's TOC, in the right order
         $parserFilename = $this->getParserFilename($indexFile, $buildContext->getHtmlOutputDir());
         $meta           = $this->getMeta($environments, $parserFilename);
-        dump(current($meta->getTocs()));
+        $files          = current($meta->getTocs());
+        array_unshift($files, sprintf('%s/index', $buildContext->getParseSubPath()));
```

Comment:
At this point `$meta->getTocs()` would contain ALL the Tocs right? Including things not in the sub-path? 

And then we prepend `$files` with the parse-path index file? I'm missing something here...

Reply:
not ALL... only the toc from the `parseSubPath`

Here we're grabbing the index.html from the subPath
```php
$basePath  = sprintf('%s/%s', $buildContext->getHtmlOutputDir(), $buildContext->getParseSubPath());
$indexFile = sprintf('%s/%s', $basePath, 'index.html');
```

Then, here we're grabbing all files from this index
```php
$parserFilename = $this->getParserFilename($indexFile, $buildContext->getHtmlOutputDir());
$meta           = $this->getMeta($environments, $parserFilename);
$files          = current($meta->getTocs());
```

And because we want the index file to be included, we're prepending it to the array:
```php
array_unshift($files, sprintf('%s/index', $buildContext->getParseSubPath()));
```